File: Templates/TypeEditor/src/Templates/Templates/VB.TypeEditor/CustomControlLibrary.Protocol/Endpoints/CustomTypeEditorOKClickEndpoint.vb
Imports System.Composition
Imports Microsoft.DotNet.DesignTools.Protocol.DataPipe
Imports Microsoft.DotNet.DesignTools.Protocol.Endpoints

Namespace Endpoints
    ''' <summary>
    '''  Endpoint to handle the event when the user clicks the OK button of the custom type editor.
    ''' </summary>
    <[Shared]>
    <ExportEndpoint>
    Public Class CustomTypeEditorOKClickEndpoint
        Inherits Endpoint(Of CustomTypeEditorOKClickRequest, CustomTypeEditorOKClickResponse)

        Public Overrides ReadOnly Property Name() As String
            Get
                Return EndpointNames.CustomTypeEditorEditorOKClick
            End Get
        End Property

        Protected Overrides Function CreateRequest(ByVal reader As IDataPipeReader) As CustomTypeEditorOKClickRequest
            Return New CustomTypeEditorOKClickRequest(reader)
        End Function

        Protected Overrides Function CreateResponse(ByVal reader As IDataPipeReader) As CustomTypeEditorOKClickResponse
            Return New CustomTypeEditorOKClickResponse(reader)
        End Function
    End Class
End Namespace

File: Templates/TypeEditor/src/Templates/Templates/CS.TypeEditor/CustomControlLibrary.Protocol/Endpoints/CustomTypeEditorOKClickEndpoint.cs
using System.Composition;
using Microsoft.DotNet.DesignTools.Protocol.DataPipe;
using Microsoft.DotNet.DesignTools.Protocol.Endpoints;

namespace CustomControlLibrary.Protocol.Endpoints
{
    /// <summary>
    ///  Endpoint to handle the event when the user clicks the OK button of the custom type editor.
    /// </summary>
    [Shared]
    [ExportEndpoint]
    public class CustomTypeEditorOKClickEndpoint : Endpoint<CustomTypeEditorOKClickRequest, CustomTypeEditorOKClickResponse>
    {
        public override string Name => EndpointNames.CustomTypeEditorEditorOKClick;

        protected override CustomTypeEditorOKClickRequest CreateRequest(IDataPipeReader reader)
            => new(reader);

        protected override CustomTypeEditorOKClickResponse CreateResponse(IDataPipeReader reader)
            => new(reader);
    }
}
